### PR TITLE
PP-10335: Send job release metric to Hosted graphite

### PR DESCRIPTION
**Pull request by Rkotyank**
added task to send job release metric from e2e-helpers

### CODE REVIEW

File: ci/tasks/send-job-release-info-to-hosted-graphite.yml
Comment (katstevens) [inline ci/tasks/send-job-release-info-to-hosted-graphite.yml:19]: We'll want to include the pipeline and job into the metric name, instead of `ENVIRONMENT` and `APP_NAME`. These should be available from the [list of built-in environment variables](https://concourse-ci.org/implementing-resource-types.html#resource-metadata) that Concourse provides.

We also need to call this task from the pipeline file above (and make sure that either `RELEASE_NUMBER` is provided, or we change the shell script here to use something else for our numeric value).

**Comment by katstevens** [comment]
Usually when we create PRs with changes to Concourse pipelines, we include a link to a successful run of that pipeline.

So you want be able to test out this pipeline and see if it’s working or not before merging :smile_cat:  The fly set-pipeline command (some [examples in the team manual here](https://pay-team-manual.cloudapps.digital/manual/tools/how-to-concourse.html#adding-a-new-pipeline)) will apply the changes from your local branch. You'll want to run something like

```
fly --target pay-dev set-pipeline --config ci/pipelines/e2e-helpers.yml --pipeline e2e-helpers
```

However, because you have a new task file, you'll also need to (temporarily) point the Git resource on the e2e helpers pipeline YAML to your branch (`PP-10335-send-e2e-helpers-job-metric-to-hosted-graphite`) instead of `master`. You don't have to commit this, just have it there while you test your pipeline. This will make sure your new file is included.

File: ci/pipelines/e2e-helpers.yml
Comment (katstevens) [inline ci/pipelines/e2e-helpers.yml:682]: We need a `.yml` on the end here 😸

File: ci/tasks/send-job-release-info-to-hosted-graphite.yml
Comment (katstevens) [inline ci/tasks/send-job-release-info-to-hosted-graphite.yml:19]: A couple of suggestions 

- try with curly braces around the `BUILD_PIPELINE_NAME` and `BUILD_JOB_NAME`
- replace the `releaseNumber` string at the end with something more descriptive for our metric, like `jobStarted` maybe?

File: ci/pipelines/e2e-helpers.yml
Comment (katstevens) [inline ci/pipelines/e2e-helpers.yml:430]: I think the `task` and the `load_var` here will need to come out of its parent `in_parallel`, because we need to wait for the `get: reverse-proxy-git-release` to complete before we can parse its tag. So just de-indent this and it should work.

File: ci/pipelines/e2e-helpers.yml
Comment (katstevens) [inline ci/pipelines/e2e-helpers.yml:335]: As per the `build-and-push-reverse-proxy-candidate` job below, these `task: parse-candidate` and `load_var` steps will need to be de-indented to come out of the `in_parallel` group above them. The `parse-candidate-tag` needs to wait for the `get: endtoend-candidate-ecr-registry-test` step to finish before it can do anything, so it can't run in parallel with it.

File: ci/pipelines/e2e-helpers.yml
Comment (katstevens) [inline ci/pipelines/e2e-helpers.yml:528]: This should be fine though - not sure why the pipeline is orange for `reverse-proxy-e2e` (unless this just needs another re-run?)

File: ci/pipelines/e2e-helpers.yml
Comment (katstevens) [inline ci/pipelines/e2e-helpers.yml:692]: As per the comments above - these `task: parse-candidate-tag` and `load_var` steps will need to be de-indented

**Comment by katstevens** [comment]
Overall this is shaping up really nicely @Rkotyank - just a few tweaks needed then I think we're ready to merge.

File: ci/pipelines/e2e-helpers.yml
Comment (katstevens) [inline ci/pipelines/e2e-helpers.yml:56]: One last thing to do before we merge this - change this back to `master` 😸

Comment (Rkotyank) [reply ci/pipelines/e2e-helpers.yml]: Done :)

**Review by katstevens — APPROVED** [review]
Brilliant stuff 🎉 I think this is a good PR to do a squash-and-merge?

**Pull request merged**
